File: backends/kotlin-ktor/src/main/kotlin/com/authorization/usecase/gate/Dto.kt
/*
 * ゲートユースケース DTO モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.usecase.gate

/**
 * JWT 発行 DTO です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class IssueDto(
    val accessToken: String,
    val memberId:    String,
)

/**
 * JWT 検証 DTO です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class VerifyDto(
    val identifier: String,
    val token:      String,
)

File: backends/rust-axum/src/usecase/gate/dto.rs
//! Gate ユースケース DTO モジュール。
//!
//! # Author
//! Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

/// JWT 発行入力 DTO。
pub struct IssueDto {
    pub access_token: String,
    pub member_id:    String,
}

/// JWT 検証入力 DTO。
pub struct VerifyDto {
    pub identifier: String,
    pub token:      String,
}
